modules/gcp/compute.go

```diff
@@ -7,146 +7,259 @@ import (
+func (i *Image) DeleteImageE(t *testing.T) error {
+	logger.Logf(t, "Destroying Image %s", i.Name)
 
 	ctx := context.Background()
-
 	service, err := NewComputeServiceE(t)
 	if err != nil {
 		return err
 	}
 
-	if _, err := service.Images.Delete(projectID, imageID).Context(ctx).Do(); err != nil {
-		return fmt.Errorf("Images.Delete(%s) got error: %v", imageID, err)
+	if _, err := service.Images.Delete(i.projectID, i.Name).Context(ctx).Do(); err != nil {
+		return fmt.Errorf("Images.Delete(%s) got error: %v", i.Name, err)
 	}
 
 	return err
 }
 
-// AddLabelsToInstance adds the tags to the given taggable instance.
-func AddLabelsToInstance(t *testing.T, projectID string, zone string, instance string, labels map[string]string) {
-	err := AddLabelsToInstanceE(t, projectID, zone, instance, labels)
+// GetInstanceIds gets the IDs of Instances in the given Instance Group.
+func (ig *ZonalInstanceGroup) GetInstanceIds(t *testing.T) []string {
+	ids, err := ig.GetInstanceIdsE(t)
 	if err != nil {
 		t.Fatal(err)
 	}
+	return ids
 }
 
-// AddLabelsToInstanceE adds the tags to the given taggable instance.
-func AddLabelsToInstanceE(t *testing.T, projectID string, zone string, instance string, labels map[string]string) error {
-	logger.Logf(t, "Adding labels to instance %s in zone %s", instance, zone)
+// GetInstanceIdsE gets the IDs of Instances in the given Zonal Instance Group.
+func (ig *ZonalInstanceGroup) GetInstanceIdsE(t *testing.T) ([]string, error) {
+	logger.Logf(t, "Get instances for Zonal Instance Group %s", ig.Name)
 
 	ctx := context.Background()
-
 	service, err := NewComputeServiceE(t)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	// Get the fingerprint of the existing labels
-	existingInstance, err := service.Instances.Get(projectID, zone, instance).Context(ctx).Do()
-	if err != nil {
-		return fmt.Errorf("Instances.Get(%s) got error: %v", instance, err)
+	requestBody := &compute.InstanceGroupsListInstancesRequest{
+		InstanceState: "ALL",
 	}
-	req := compute.InstancesSetLabelsRequest{Labels: labels, LabelFingerprint: existingInstance.LabelFingerprint}
 
-	// Perform the SetLabels request
-	if _, err := service.Instances.SetLabels(projectID, zone, instance, &req).Context(ctx).Do(); err != nil {
-		return fmt.Errorf("Instances.SetLabels(%s) got error: %v", instance, err)
+	instanceIDs := []string{}
+	zone := ZoneUrlToZone(ig.Zone)
+
+	req := service.InstanceGroups.ListInstances(ig.projectID, zone, ig.Name, requestBody)
+	if err := req.Pages(ctx, func(page *compute.InstanceGroupsListInstances) error {
+		for _, instance := range page.Items {
+			// For some reason service.InstanceGroups.ListInstances returns us a collection
+			// with Instance URLs and we need only the Instance ID for the next call. Use
+			// the path functions to chop the Instance ID off the end of the URL.
+			instanceID := path.Base(instance.Instance)
+			instanceIDs = append(instanceIDs, instanceID)
+		}
+		return nil
+	}); err != nil {
```

Comment:
The `if err := xxx; err != nil` idiom works well for short amounts of code, but not for code with a multi-line function defined in `xxx`. Please break into multiple lines.